java/arcs/core/storage/DriverFactory.kt

```diff
@@ -57,24 +54,19 @@ object DriverFactory {
      * instances. Therefore using this method requires shutting down all arcs, and should be use
      * only in rare circumstances.
      */
-    suspend fun removeAllEntities(): Job = coroutineScope {
-        launch {
-            providers.value.forEach { it.removeAllEntities() }
-        }
+    suspend fun removeAllEntities() {
+        providers.value.forEach { it.removeAllEntities() }
     }
 
     /**
      * Clears all entities created in the given time range. See comments on [removeAllEntities] re
      * the need to recreate stores after calling this method.
      */
-    suspend fun removeEntitiesCreatedBetween(startTimeMillis: Long, endTimeMillis: Long): Job =
-        coroutineScope {
-            launch {
-                providers.value.forEach {
-                    it.removeEntitiesCreatedBetween(startTimeMillis, endTimeMillis)
-                }
-            }
+    suspend fun removeEntitiesCreatedBetween(startTimeMillis: Long, endTimeMillis: Long) {
+        providers.value.forEach {
+            it.removeEntitiesCreatedBetween(startTimeMillis, endTimeMillis)
```

Comment:
Does it make sense to launch a coroutine for each of the providers in each of these methods, to perform the operations in parallel? I guess neither of these are all that performance sensitive, so it's probably fine either way. Just a thought.

Reply:
It does, not just for performance reasons, but for the stated of having one provider failure not result in the rest of them failing.

Reply:
Argh, your comment confused me because I thought it's what I was already doing, and I realized I hadn't committed my changes before pushing and requesting re-review :(